service/debugger/debugger.go

```diff
@@ -149,7 +150,18 @@ func (d *Debugger) CreateBreakpoint(requestedBp *api.Breakpoint) (*api.Breakpoin
 	)
 	switch {
 	case len(requestedBp.File) > 0:
-		addr, err = d.process.FindFileLocation(normalizePath(requestedBp.File), requestedBp.Line)
+		fileName := requestedBp.File
+		if runtime.GOOS == "windows" {
```

Comment:
> Instead of adding the check here, let's remove the assumption in  FindFileLocation  that its input has been normalized, let's take that burden off the caller and centralize that logic within that method. That way we're not special casing throughout the codebase.

Happy to do this.  Though there is only one caller currently who has to worry about this, so making this change won't reduce the number of places in total, it will just push the boundary of where path normalization further in.  I think that's fine though, and make this a little less brittle.

It does though mean that we will need to duplicate `normalizePath` since we'll have users of that functionality in both `service/debugger` and `proc`, and there's no clean way I can see to share utility code like `normalizePath` between these two.

>> Additionally, since  normalizePath  already is checking  runtime.GOOS  let's remove that from any logic that calls it. 

In that case, we would do this key iteration on all platforms, which will be less efficient on platforms that don't need it, but is otherwise okay.

> why can't we just assume that the string returned from  normalizePath  will correctly match a file (if it exists in sources)?

There's no good way I can tell to correctly guess what normalization the Go compiler will have used.  That's actually exactly the problem that this PR is fixing.

I had previously thought that the Go compiler always did a normalization to lowercase with `/`.  But I was only seeing that because I had my `GOPATH` set to lowercase and happened to be using all lower case files.  

So to do the case-insensitive path comparison we actually have to normalize both the requested filename as well as the name used as a key in the Go symbol table.  The latter is built outside of Delve's control by the Go compiler - potentially even on a different machine.

As an example, imagine that the Go compiler embedded `C:\go\src\foo.go`, but the user requests a bp on `c:\Go\src\foo.go`.  These should match, but neither one normalizes to the other.